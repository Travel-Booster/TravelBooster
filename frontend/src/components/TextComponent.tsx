import React from 'react'
import message from '../services/i18n/i18-js'
import { Text } from 'react-native'

type TextComponentProps = {
  text: string
}

export const TextComponent = React.memo<TextComponentProps>(({ text }) => {
  return <Text>{message(text)}</Text>
})
