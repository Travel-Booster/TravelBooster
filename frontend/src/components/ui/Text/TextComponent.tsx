import React, { memo } from 'react'
import { Text } from 'react-native'
import { message } from "helpers/getTranslation"

type TextComponentProps = {
  text: string
}

/**
 * @param {string}  p1 - A string param.
 * @param {string=} p2 - An optional param (Closure syntax)
 * @param {string} [p3] - Another optional param (JSDoc syntax).
 * @param {string} [p4="test"] - An optional param with a default value
 * @return {string} This is the result
 */
export const TextComponent = memo<TextComponentProps>(({ text }) => {
  return <Text>{message(text)}</Text>
})
