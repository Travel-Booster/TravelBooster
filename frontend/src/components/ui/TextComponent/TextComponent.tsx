import React, { memo } from 'react'
import { Text } from 'react-native'
import { message } from 'helpers/getTranslation'

type TextComponentProps = {
  text: string
}
/**
 * @namespace TextComponent
 * @memberof app.components
 */
const TextComponent = memo<TextComponentProps>(({ text }) => {
  return <Text>{message(text)}</Text>
})

export default TextComponent
