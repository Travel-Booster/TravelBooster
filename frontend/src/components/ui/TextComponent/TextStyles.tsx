import { StyleSheet } from 'react-native'
import colors from 'styles/colors'

export default StyleSheet.create({
  normal: { 
    fontFamily: 'montserrat-regular',
    fontSize: 16,
    color: colors.LIGHT_GRAY_LIGHT
  },
  heading: { 
    fontFamily: 'montserrat-extra-bold',
    fontSize: 36,
    color: colors.LIGHT_GRAY_DARK
  },
  sub: { 
    fontFamily: 'montserrat-semi-bold',
    fontSize: 18,
    color: colors.LIGHT_GRAY_DARK
  },
  title: { 
    fontFamily: 'montserrat-semi-bold',
    fontSize: 16,
    color: colors.LIGHT_GRAY_DARK
  },
  description: { 
    fontFamily: 'montserrat-regular',
    fontSize: 12,
    color: colors.LIGHT_GRAY_LIGHT
  },
  button: { 
    fontFamily: 'montserrat-semi-bold',
    fontSize: 14,
    color: colors.LIGHT_MAIN
  }
})
