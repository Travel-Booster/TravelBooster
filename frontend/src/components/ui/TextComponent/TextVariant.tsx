import colors from 'services/styles/colors'

export const normalTextStyles = { 
 style: { 
   fontFamily: 'Montserrat_400Regular',
   fontSize: 16,
   color: colors.LIGHT_GRAY_LIGHT
 }
}
 
export const buttonTextStyles = { 
 style: { 
   fontFamily: 'Montserrat_600SemiBold',
   fontSize: 14,
   color: colors.LIGHT_MAIN
 }
}
