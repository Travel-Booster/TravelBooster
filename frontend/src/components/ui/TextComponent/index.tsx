/**
 * export this file
 * @namespace getTranslation
 * @memberof app.components
 */
export { default } from './TextComponent'
