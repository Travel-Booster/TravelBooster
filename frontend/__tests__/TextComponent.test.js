import React from 'react'
import renderer from 'react-test-renderer'
import TextComponent from 'components/ui/TextComponent'

const props = {
    text: 'hello',
    variant: 'button'
}

const component = <TextComponent {...props} />
let wrapper = renderer.create(component).toJSON()

it('<TextComponent /> renders', () => {
    expect(wrapper).toBeTruthy()
})
    
it('<TextComponent /> test snapshot', () => {
    expect(wrapper).toMatchSnapshot()
})
