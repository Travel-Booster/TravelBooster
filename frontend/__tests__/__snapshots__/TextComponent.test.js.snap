// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> test snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#FFFFFF",
      "fontFamily": "montserrat-semi-bold",
      "fontSize": 14,
    }
  }
>
  Hello world
</Text>
`;
